src/virtualship/cli/commands.py

```diff
@@ -45,9 +47,76 @@ def init(path):
     "path",
     type=click.Path(exists=True, file_okay=False, dir_okay=True, readable=True),
 )
-def fetch(path):
-    """Entrypoint for the tool."""
-    raise NotImplementedError("Not implemented yet.")
+def fetch(expedition_dir: str | Path) -> None:
+    """Entrypoint for the tool to download data based on area of interest."""
+    if isinstance(expedition_dir, str):
+        expedition_dir = Path(expedition_dir)
+
+    # Load schedule
+    schedule = _get_schedule(expedition_dir)
+    if schedule is None:
+        print("Error: Schedule file not found.")
+        return
+
+    # Extract area_of_interest details from the schedule
+    spatial_range = schedule.area_of_interest.spatial_range
+    time_range = schedule.area_of_interest.time_range
+    start_datetime = datetime.strptime(time_range.start_time, "%Y-%m-%d %H:%M:%S")
+    end_datetime = datetime.strptime(time_range.end_time, "%Y-%m-%d %H:%M:%S")
+
+    # Prompt for user credentials
+    username = input("username: ")
+    password = input("password: ")
```

Comment:
Let's use click to field these inputs

```diff
diff --git a/src/virtualship/cli/commands.py b/src/virtualship/cli/commands.py
index f533b96..800eb0f 100644
--- a/src/virtualship/cli/commands.py
+++ b/src/virtualship/cli/commands.py
@@ -47,7 +47,13 @@ def init(path):
     "path",
     type=click.Path(exists=True, file_okay=False, dir_okay=True, readable=True),
 )
-def fetch(expedition_dir: str | Path) -> None:
+@click.option(
+    "--username", prompt=True, type=str,
+)
+@click.option(
+    "--password", prompt=True, hide_input=True, type=str,
+)
+def fetch(expedition_dir: str | Path, username: str, password: str) -> None:
     """Entrypoint for the tool to download data based on area of interest."""
     if isinstance(expedition_dir, str):
         expedition_dir = Path(expedition_dir)
@@ -64,10 +70,6 @@ def fetch(expedition_dir: str | Path) -> None:
     start_datetime = datetime.strptime(time_range.start_time, "%Y-%m-%d %H:%M:%S")
     end_datetime = datetime.strptime(time_range.end_time, "%Y-%m-%d %H:%M:%S")
 
-    # Prompt for user credentials
-    username = input("username: ")
-    password = input("password: ")
-
     # Define all datasets to download, including bathymetry
     download_dict = {
         "Bathymetry": {

```

Reply:
Great idea! And also support credientials stored in a file like in https://help.marine.copernicus.eu/en/articles/8185007-copernicus-marine-toolbox-credentials-configuration?

Reply:
And perhaps also provide a (short) message how to obtain these credentials? Via https://data.marine.copernicus.eu/register?

Reply:
Done in c691a478fea5443d3680152c5c1e0c1e9bc6f14c